src/utils/metrics/arg_parsers.rs

```diff
@@ -0,0 +1,27 @@
+use anyhow::{anyhow, Result};
+
+// Parse the value argument for a set metric.
+pub fn set_value_parser(s: &str) -> Result<i64> {
+    match s.parse::<i64>() {
+        Ok(res) => Ok(res),
+        Err(_) => match s.parse::<f64>() {
+            Ok(_) => Err(anyhow!(format!("floats are not supported for set metrics"))),
+            Err(_) => Ok(crc32fast::hash(s.as_bytes()) as i64),
+        },
+    }
```

Comment:
I think two things:
  - We should just treat floating point values as strings
  - It might be a good idea to force users to specify the type of data they are sending. Otherwise, a user who sends a string containing only digits will have that string treated as an int